h2o-core/src/main/java/water/Value.java

```diff
@@ -268,7 +268,13 @@ public static String nameOfPersist(int x) {
   public boolean isKey()      { return _type != TypeMap.PRIM_B  && TypeMap.theFreezable(_type) instanceof Key; }
   /** Check if the Value's POJO is a {@link Frame} subtype.  Does not require the POJO.
    *  @return True if the Value's POJO is a {@link Frame} subtype. */
-  public boolean isFrame()    { return _type != TypeMap.PRIM_B  && TypeMap.theFreezable(_type) instanceof Frame; }
+  public boolean isFrame()    {
+      try {
+          return _type != TypeMap.PRIM_B  && TypeMap.getTheFreezableOrThrow(_type) instanceof Frame;
+      } catch (ClassNotFoundException e) {
```

Comment:
This is a controversial change. However, the previous implementation might lead to failures. The sole purpose of `isSomething` method is to determine if it is something or not. If the give class can not be found in the pre-defined type map and the search boundary is withing pre-defined map, then it certainly **is not** what we're asking it to be.

Reply:
How is it possible that we can get a class not found exception?

Reply:
BAD is not an actual class, let me try to reproduce and make a suggestion